src/components/admin/PortalAddEditForm.tsx

```diff
@@ -113,8 +159,24 @@ export const PortalAddEditForm: React.FC<PortalAddEditFormProps> = ({
   const changeRoomImageUrl = useCallback(
     ({ url }) => {
       setValue("image_url", url, { shouldValidate: false });
+
+      // Get the new image and use it to calculate new width/height for the
+      // image so that the aspect ratio isn't wrong.
+      const checkImage = new Image();
+      checkImage.src = url;
+      checkImage.decode().then(() => {
+        const [newWidth, newHeight] = determineNewWidthAndHeight({
+          portal,
+          newWidthPx: checkImage.naturalWidth,
+          newHeightPx: checkImage.naturalHeight,
+          mapWidthPx,
+          mapHeightPx,
+        });
+        setValue("width_percent", newWidth);
+        setValue("height_percent", newHeight);
+      });
```

Comment:
I recommend not leaving uncaught rejections. Attaching `.catch()` here akin to https://github.com/sparkletown/sparkle/blob/redesign-2021/src/components/templates/Auditorium/components/Section/Section.tsx#L47-L53
would make it useful to know if a problem ocurred:

```suggestion
      }).catch(
        createErrorCapture({
          message: "Problem setting width and height percents",
          where: "PortalAddEditForm",
        })
      );
```